### PR TITLE
More hqDefine

### PR DESCRIPTION
**Pull request by orangejenny**
@gcapalbo / @millerdev 

Review by commit.

### CODE REVIEW

File: corehq/apps/appstore/static/appstore/js/facet_sidebar.js [hunk 1]
```diff
@@ -1,101 +1,103 @@
...
-}
-
-function replaceAll(find, replace, str) {
-    // http://stackoverflow.com/questions/1144783/replacing-all-occurrences-of-a-string-in-javascript#answer-1144788
-    return str.replace(new RegExp(escapeRegExp(find), 'g'), replace);
-}
+hqDefine("appstore/js/facet_sidebar", function() {
+    var escapeRegExp = function(str) {
+        return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
+    }
```
[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/apps/appstore/static/appstore/js/facet_sidebar.js]: You changed the way the function was defined, so better to either change it back or add a semicolon to remove the lint error.

[the remark this answers is not in the document]
Comment (millerdev) [reply corehq/apps/appstore/static/appstore/js/facet_sidebar.js]: You changed the way the function was defined, so better to either change it back or add a semicolon to remove the lint error.

**Review by millerdev — APPROVED** [review]

File: corehq/apps/hqadmin/static/hqadmin/js/visualizations.js [hunk 2]
```diff
@@ -1,195 +1,203 @@
...
 
-    function _update_chart_if_exists(chart) {
-        if (chart) {
-            chart.update();
+    // adapted from http://stackoverflow.com/questions/901115/
+    var getUrlParams = function(a) {
+        // takes a string of the form "foo=bar&bat=&fizz=bang"
+        // and returns {"foo": "bar", "bat": "", "fizz": "bang"}
+        a = a.split('&')
+        if (a == "") return {};
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/hqadmin/static/hqadmin/js/visualizations.js]: This wasn't doing anything; `split` always returns an array.

File: corehq/apps/appstore/static/appstore/js/facet_sidebar.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/appstore/static/appstore/js/facet_sidebar.js]: updated

**Review by millerdev — APPROVED** [review]

**Pull request merged**